.travis.yml

```diff
@@ -189,28 +189,32 @@ before_install:
           fi
 
           if [[ $PHP == 7.3 ]]; then
-              # System libmemcached is missing sasl
-              LIBMEMCACHED_VERSION=1.0.18
-              wget "https://launchpad.net/libmemcached/1.0/${LIBMEMCACHED_VERSION}/+download/libmemcached-${LIBMEMCACHED_VERSION}.tar.gz" -O libmemcached-${LIBMEMCACHED_VERSION}.tar.gz
-              tar xvfz libmemcached-${LIBMEMCACHED_VERSION}.tar.gz
-              cd "libmemcached-${LIBMEMCACHED_VERSION}"
-              mkdir -p /opt/libmemcached/${LIBMEMCACHED_VERSION}
-              ./configure --prefix="/opt/libmemcached/${LIBMEMCACHED_VERSION}" LDFLAGS="-lpthread"
-              make
-              make install
-              cd -
-
-              EXTMEMCACHED_VERSION=3.1.3
-              wget https://pecl.php.net/get/memcached-${EXTMEMCACHED_VERSION}.tgz
-              tar xzvf memcached-${EXTMEMCACHED_VERSION}.tgz
-              cd memcached-${EXTMEMCACHED_VERSION}
-              phpize
-              ./configure --with-libmemcached-dir=/opt/libmemcached/${LIBMEMCACHED_VERSION}
-              make
-              sudo make install
-              cd -
+              ext_dir=$(php -r "echo ini_get('extension_dir');")
+              ext_cache=~/php-ext/$(basename $ext_dir)/memcached.so
+
+              if [[ ! -e $ext_cache ]]; then
+                  # System libmemcached is missing sasl
+                  LIBMEMCACHED_VERSION=1.0.18
+                  wget "https://launchpad.net/libmemcached/1.0/${LIBMEMCACHED_VERSION}/+download/libmemcached-${LIBMEMCACHED_VERSION}.tar.gz" -O libmemcached-${LIBMEMCACHED_VERSION}.tar.gz
+                  tar xvfz libmemcached-${LIBMEMCACHED_VERSION}.tar.gz
+                  cd "libmemcached-${LIBMEMCACHED_VERSION}"
+                  mkdir -p /opt/libmemcached/${LIBMEMCACHED_VERSION}
+                  ./configure --prefix="/opt/libmemcached/${LIBMEMCACHED_VERSION}" LDFLAGS="-lpthread"
+                  make
+                  make install
+                  cd -
+
+                  EXTMEMCACHED_VERSION=3.1.3
+                  wget https://pecl.php.net/get/memcached-${EXTMEMCACHED_VERSION}.tgz
+                  tar xzvf memcached-${EXTMEMCACHED_VERSION}.tgz
+                  cd memcached-${EXTMEMCACHED_VERSION}
+                  phpize
+                  ./configure --with-libmemcached-dir=/opt/libmemcached/${LIBMEMCACHED_VERSION}
+                  make
+                  cp modules/memcached.so $ext_cache
+                  cd -
+              fi
           fi
-
           echo extension = memcached.so >> $INI
```

Comment:
ok, got it, should be inside the previous if (but not inside the nested if), isn't it?